source/MRMesh/MRVector.h

```diff
@@ -110,21 +110,21 @@ class Vector
     void pop_back() { vec_.pop_back(); }
 
     template<typename... Args>
-    decltype(auto) emplace_back( Args&&... args ) { return vec_.emplace_back( std::forward<Args>(args)... ); }
+    T& emplace_back( Args&&... args ) { return vec_.emplace_back( std::forward<Args>(args)... ); }
 
     [[nodiscard]] const_reference front() const { return vec_.front(); }
     [[nodiscard]]       reference front()       { return vec_.front(); }
     [[nodiscard]] const_reference  back() const { return vec_.back(); }
     [[nodiscard]]       reference  back()       { return vec_.back(); }
     /// returns the identifier of the first element
-    [[nodiscard]] I beginId() const { return I{ size_t(0) }; }
+    [[nodiscard]] I beginId() const { return I( size_t(0) ); }
     /// returns the identifier of the back() element
-    [[nodiscard]] I backId() const { assert( !vec_.empty() ); return I{ vec_.size() - 1 }; }
+    [[nodiscard]] I backId() const { assert( !vec_.empty() ); return I( vec_.size() - 1 ); }
     /// returns backId() + 1
-    [[nodiscard]] I endId() const { return I{ vec_.size() }; }
+    [[nodiscard]] I endId() const { return I( vec_.size() ); }
 
-    [[nodiscard]] auto data() { return vec_.data(); }
-    [[nodiscard]] auto data() const { return vec_.data(); }
+    [[nodiscard]] T* data() { return vec_.data(); }
+    [[nodiscard]] const T* data() const { return vec_.data(); }
```

Comment:
we cant use auto anymore?

Reply:
You still can. I just don't want it in this particular case, because `data()` can return a pointer to some typedef (`value_type` or something). I'd rather control the exact return type.

Reply:
Since typedefs become different types on python side (possibly not even convertible to each other, though I might need to change that), not purely cosmetical at the moment.